### ISSUE TITLE
C++20 support

### ISSUE THREAD

**Issue by hly1204** [yosupo06/library-checker-problems#735]
Would you like to add the C++20 option? I think the latest version of GNU compiler has most new features done.

**Comment by yosupo06** [comment]
> Would you like to add the C++20 option? I think the latest version of GNU compiler has most new features done.

Yes, but the gcc version of our judge is not so new (9.3.0, it is the default one of ubuntu), so C++20 may not be supported fully

**Comment by learningstud** [comment]
As outlined in the official documentation for [toolchain updates](https://wiki.ubuntu.com/ToolChain#Toolchain_Updates), it seems possible to update gcc with [ppa:ubuntu-toolchain-r](https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/ppa) which has a gcc12 build. Would this be feasible?

**Comment by yosupo06** [comment]
Recently I updated the gcc into 12.1 (https://github.com/yosupo06/library-checker-judge/blob/master/langs/Dockerfile.GCC ). In my understanding, it (almost) fully support C++20, so I'll add C++20
